### PR TITLE
Add spring integration chapter

### PR DESCRIPTION
**Pull request by gurbuzali**

### CODE REVIEW

File: docs/070_Configuration/020_Spring_Integration.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+You can integrate Hazelcast Jet with Spring and this chapter explains 
+the configuration of Hazelcast Jet within Spring context.
+ 
+
+## Declaring Beans by Spring beans Namespace
+
+You can declare Hazelcast Jet Objects using the default Spring beans namespace.
+Example code for a Hazelcast Jet Instance declaration is listed below.
```
Comment (Serdaro) [inline docs/070_Configuration/020_Spring_Integration.md]: Here is an example code for a Hazelcast Jet Instance declaration:

File: docs/070_Configuration/020_Spring_Integration.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+ - map
+ - list
+ 
+ ```xml
+<jet:map instance-ref="jet-instance" name="my-map" id="my-map-bean"/>
+
+<jet:list instance-ref="jet-client" name="my-list" id="my-list-bean"/>
+```
+
+You can obtain underlying `HazelcastInstance` as a bean and use this bean to
```
Comment (Serdaro) [inline docs/070_Configuration/020_Spring_Integration.md]: ..obtain the underlying..

File: docs/070_Configuration/020_Spring_Integration.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+</jet:instance>
+<jet:client id="client" scope="prototype" depends-on="instance">
+<!-- ... -->
+</jet:client>
+```
+
+## Annotation-Based Configuration 
+
+Annotation-Based Configuration does not require any XML definition. Simply create 
+a class with related annotations, eg `@Configuration`. And provide JetInstance as a
```
Comment (Serdaro) [inline docs/070_Configuration/020_Spring_Integration.md]: ...annotations, e.g., `@Configuration` and provide...

File: docs/070_Configuration/020_Spring_Integration.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+You can mark your custom processors with `@SpringAware` annotation which 
+gives you the ability:
+
+ - to apply bean properties
+ - to apply factory callbacks such as `ApplicationContextAware`, `BeanNameAware`
+ - to apply bean post-processing annotations such as `InitializingBean`, `@PostConstruct`
+ 
+ Refer to 
+ [code samples](https://github.com/hazelcast/hazelcast-jet-code-samples/blob/master/spring/src/main/java/jet/spring/source/CustomSourceP.java)
+ for more information
```
Comment (Serdaro) [inline docs/070_Configuration/020_Spring_Integration.md]: ..more information.

File: docs/070_Configuration/020_Spring_Integration.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        <hz:member>127.0.0.1:5701</hz:member>
+    </jet:network>
+    <jet:spring-aware/>
+</jet:client>
+```
+
+#### Hazelcast Jet Supported Type Configurations and Examples
+
+ - map
+ - list
```
Comment (vladoschreiner) [inline docs/070_Configuration/020_Spring_Integration.md]: Why not Cache? ICache is supported accros Jet as well.

File: docs/070_Configuration/020_Spring_Integration.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+[annotation-based](https://github.com/hazelcast/hazelcast-jet-code-samples/blob/master/spring/src/main/java/jet/spring/AnnotationBasedConfigurationSample.java)
+configurations are available at our Code Samples repo.
+
+You can mark your custom processors with `@SpringAware` annotation which 
+gives you the ability:
+
+ - to apply bean properties
+ - to apply factory callbacks such as `ApplicationContextAware`, `BeanNameAware`
+ - to apply bean post-processing annotations such as `InitializingBean`, `@PostConstruct`
+
```
Comment (vladoschreiner) [inline docs/070_Configuration/020_Spring_Integration.md]: I'd prefer to first explain what is SpringAware, then how to configure it.

Also, I'd add an introduction. Roughly: Jet job is submitted to cluster, cluster has it's own Spring context. This is the way how to make the context accessible. 
Then the co

File: docs/070_Configuration/020_Spring_Integration.md (hunk 5 above)
Comment (gurbuzali) [reply docs/070_Configuration/020_Spring_Integration.md]: Hazelcast IMDG does not provide direct access to the ICache as well instead exposes it with `CacheManager` interface and integrates it with Spring Cache Api. I've chosen the same approach for Jet.

**Comment by gurbuzali** [comment]
@cangencer https://github.com/hazelcast/hazelcast-jet-reference-manual/pull/68

**Comment by gurbuzali** [comment]
closing in favor of https://github.com/hazelcast/hazelcast-jet-reference-manual/pull/68

**Pull request closed without merging**